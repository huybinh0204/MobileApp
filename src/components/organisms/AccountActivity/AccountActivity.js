import { useTheme } from '@emotion/react';
import { useFocusEffect, useNavigation } from '@react-navigation/native';
import { DateTime } from 'luxon';
import React, { useCallback, useMemo } from 'react';
import { ActivityIndicator } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { MainButton, TransactionItem } from '_components/atoms';
import { ListCard, ListItem } from '_components/molecules';
import { NAVIGATION } from '_constants';
import strings from '_localization';
import { AccountActions, AccountSelectors } from '_store/account';
import { ButtonContainer } from './AccountActivity.styles';

const AccountActivity = () => {
  const { colors } = useTheme();
  const { navigate } = useNavigation();
  const dispatch = useDispatch();

  const accountExternalId = useSelector(AccountSelectors.getExternalId);
  const isLoading = useSelector(AccountSelectors.getIsLoadingTransactions);
  const transactions = useSelector(AccountSelectors.getTransactions);

  const isTransactionsEmpty = transactions.length === 0;
  const monthToDate = useMemo(() => DateTime.now().minus({ days: 30 }).toISO(), []);

  const navigateToTransactionsList = () => {
    dispatch(AccountActions.cleanTransactions());
    navigate(NAVIGATION.accounts.transactions);
  };

  const fetchRecentTransactions = useCallback(() => {
    if (accountExternalId) {
      dispatch(
        AccountActions.fetchTransactions(accountExternalId, {
          page: 1,
          maxCount: 3,
          startDate: monthToDate,
        })
      );
    }
  }, [accountExternalId, monthToDate, dispatch]);

  useFocusEffect(fetchRecentTransactions);

  return (
    <ListCard title={strings.accounts.nudgeTitle} testID="AccountActivity">
      {isTransactionsEmpty && isLoading && <ActivityIndicator color={colors.alpha500} />}
      {isTransactionsEmpty && !isLoading ? (
        <ListItem color={colors.beta900} title={strings.accounts.nudgeEmptyState} />
      ) : (
        transactions.map((transaction, index) => (
          <TransactionItem
            key={transaction.id}
            isLastItem={index === transactions.length - 1}
            transaction={transaction}
          />
        ))
      )}
      <ButtonContainer isTransactionsEmpty={isTransactionsEmpty}>
        <MainButton onPress={navigateToTransactionsList} variant="secondary">
          {strings.accounts.nudgeActionTitle}
        </MainButton>
      </ButtonContainer>
    </ListCard>
  );
};

export default AccountActivity;
