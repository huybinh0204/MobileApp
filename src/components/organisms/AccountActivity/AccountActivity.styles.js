import styled from '@emotion/native';

export const ButtonContainer = styled.View(({ theme, isTransactionsEmpty }) => ({
  marginTop: isTransactionsEmpty ? theme.spacing.s : 0,
}));
