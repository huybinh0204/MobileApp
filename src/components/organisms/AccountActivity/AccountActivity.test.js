import { render } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import AccountActivity from './AccountActivity';

describe('AccountActivity', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={AccountActivity} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
