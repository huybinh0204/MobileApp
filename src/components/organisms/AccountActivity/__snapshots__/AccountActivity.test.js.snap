// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountActivity should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                              "backgroundColor": "#FFFFFF",
                              "borderColor": "#F2F3F4",
                              "borderRadius": 10,
                              "borderWidth": 0,
                              "padding": 29,
                              "shadowColor": "#362E60",
                              "shadowOffset": Object {
                                "height": 7,
                                "width": 0,
                              },
                              "shadowOpacity": 0.1,
                              "shadowRadius": 10,
                            },
                            Array [
                              Object {
                                "marginBottom": 29,
                              },
                              undefined,
                            ],
                          ]
                        }
                        testID="AccountActivity"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#1F1D25",
                                "flex": 1,
                                "fontFamily": "Poppins-SemiBold",
                                "fontSize": 36,
                                "lineHeight": 50.5,
                                "marginBottom": 29,
                              },
                              undefined,
                            ]
                          }
                        >
                          Recent Activity
                        </Text>
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "alignSelf": "stretch",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "flexDirection": "row",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                color="#1F1D25"
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "width": "80%",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                No activity in last 30 days
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          isTransactionsEmpty={true}
                          style={
                            Array [
                              Object {
                                "marginTop": 29,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessibilityState={
                              Object {
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": "stretch",
                                "backgroundColor": "#EBE9FD",
                                "borderRadius": 10,
                                "opacity": 1,
                                "paddingHorizontal": 45.5,
                                "paddingVertical": 31,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              disabled={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#3D3397",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              View all activity
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
